### PR TITLE
set correct 'targetEnvironment' value

### PR DESCRIPTION
**Pull request by dmitsh**

### CODE REVIEW

**Comment by brendandburns** [comment]
Can one of the admins verify this patch?

File: pkg/acsengine/engine.go [hunk 1]
```diff
@@ -295,15 +295,25 @@ func prepareTemplateFiles(properties *api.Properties) ([]string, string, error)
...
+		return AzureChinaCloudSpec
+	default:
+		return AzureCloudSpec
 	}
+}
 
-	return cloudSpecConfig
+func GetCloudTargetEnv(location string) string {
+	switch location {
+	case "chinaeast":
```
Comment (colemickens) [inline pkg/acsengine/engine.go]: You can do this too for the switches...

```
switch {
case location=="chinaeast" || location=="chinanorth":
    return AzureChinaCloud
default:
    return AzurePublicCloud
}
```

I'll let you decide if it's better or worse 😄 .

**Comment by dmitsh** [comment]
I deployed and verified all 4 orchestrators with these changes. Does it count?

Thanks,
-Dmitry

File: pkg/acsengine/engine.go (hunk 1 above)
Comment (dmitsh) [inline pkg/acsengine/engine.go]: @weinong thinks we should add strings like "China East" and "China North". Should we?

**Comment by colemickens** [comment]
@dmitsh that's your PR builder commenting, not Brendan :P

File: pkg/acsengine/engine.go (hunk 1 above)
Comment (colemickens) [inline pkg/acsengine/engine.go]: Can we canonicalize them first? Is it true that for any "Location Name" that "locationname" will always be valid?

@weinong is probably right even though I wish he weren't... if that makes sense...

File: pkg/acsengine/engine.go [hunk 2]
```diff
@@ -295,15 +300,27 @@ func prepareTemplateFiles(properties *api.Properties) ([]string, string, error)
...
+	switch GetCloudTargetEnv(location) {
+	case azureCloudChina:
+		return AzureChinaCloudSpec
+	default:
+		return AzureCloudSpec
 	}
+}
 
-	return cloudSpecConfig
+func GetCloudTargetEnv(location string) string {
```
Comment (weinong) [inline pkg/acsengine/engine.go]: there is a potential bug here. "The rest of the locations" are not public azure. there are USGov, Germany, etc... :(

Comment (dmitsh) [inline pkg/acsengine/engine.go]: addressed, thanks!

**Comment by wangtt03** [comment]
@dmitsh Thanks for your commit.😊👍

**Pull request merged**